tempesta_fw/sock_srv.c

```diff
@@ -446,7 +446,8 @@ tfw_sock_srv_disconnect(TfwConn *conn)
 	 * attached to the connection are released. If the connection is
 	 * closed already, then force the release of attached resources.
 	 */
-	if (atomic_read(&conn->refcnt) != TFW_CONN_DEATHCNT)
+	refcnt = atomic_read(&conn->refcnt);
+	if (refcnt && refcnt != TFW_CONN_DEATHCNT)
```

Comment:
Could you please describe the case when the reference count is zero? It's not obvious from the patch. A good comment would be helpful.
The patch introduces extra work in a potentially frequently executed code, while it appears that it's for a very limited case. Can it be done differently? Like, calling `tfw_connection_put_to_death()` beforehand in that particular case?

Reply:
@keshonok Thank you for review.
Fix has been reworked and extended. Comment added.